### PR TITLE
coord: plan CREATE SOURCES FROM POSTGRES

### PR DESCRIPTION
**Pull request by JLDLaughlin**
Adds the planning portion to `CREATE SOURCES FROM POSTGRES`. This codepath is still
stopped at `plan_create_sources`, so we don't insert bad sources into the catalog that prevent
us from testing.

### CODE REVIEW

File: src/sql/src/plan.rs [hunk 1]
```diff
@@ -76,11 +76,9 @@ pub enum Plan {
     CreateRole {
         name: String,
     },
-    CreateSource {
-        name: FullName,
-        source: Source,
-        if_not_exists: bool,
-        materialized: bool,
+    CreateSource(CreateSourcePlan),
```
Comment (sploiselle) [inline src/sql/src/plan.rs:79]: Why not only have the plural?

File: src/sql/src/plan/statement/ddl.rs [hunk 2]
```diff
@@ -1027,18 +1027,25 @@ pub fn plan_create_source(
...
+    }))
 }
 
 pub fn plan_create_sources(
-    _scx: &StatementContext,
-    _stmt: CreateSourcesStatement<Raw>,
+    scx: &StatementContext,
+    stmt: CreateSourcesStatement<Raw>,
 ) -> Result<Plan, anyhow::Error> {
+    scx.require_experimental_mode("Postgres Sources")?;
```
Comment (sploiselle) [inline src/sql/src/plan/statement/ddl.rs:1042]: I have to say that I am pretty lost here! I'm not sure why `plan_create_sources` gets winnowed down to only caring about Posgres sources, so I might not be the right person to review.

File: test/pg-cdc/pg-cdc.td [hunk 3]
```diff
@@ -101,15 +101,15 @@ incorrect column specification: specified column does not match upstream source,
     PUBLICATION 'mz_source_${testdrive.seed}'
     NAMESPACE 'public'
     TABLES ('numbers_${testdrive.seed}' AS "numbers")
-CREATE SOURCES not yet supported
+Postgres sources not yet supported
```
Comment (sploiselle) [inline test/pg-cdc/pg-cdc.td:104]: Nit: Keeping these error messages in the exact same case the `str` that generated them in the code base makes things easier to find. Looks like this is [`"Postgres Sources"`](https://github.com/MaterializeInc/materialize/pull/6448/files#diff-c840a7fbb6a642d6cbf71dddd92ec1b6dfa86310022ead5673e8fa509ba73dc4R1042) above.

File: src/sql/src/plan.rs (hunk 1 above)
Comment (sploiselle) [reply src/sql/src/plan.rs:79]: Reading further this seems like there is some hidden invariant driving this design?

**Review by sploiselle — COMMENTED** [review]
Sorry I don't have anything of substance to add here. If I can get a little more context on what this design is supposed to facilitate, glad to give it another pass. If Petros is on top of things, though, no need to expend the effort.

File: src/sql/src/plan/statement/ddl.rs (hunk 2 above)
Comment (JLDLaughlin) [reply src/sql/src/plan/statement/ddl.rs:1042]: You're totally right that `CREATE SOURCES` will only work with Postgres sources at the moment. [The idea](https://github.com/MaterializeInc/materialize/pull/6299#issuecomment-812102095) was that users would be able to provide the connection information for Postgres once, while specifying multiple upstream tables. I'm not sure if there's any plan to expand it to other source types in the near-term or long-term future.

**Review by petrosagg — APPROVED** [review]
looks good to me!

**Pull request merged**